content/BatchRequestContentCollection.md

```diff
@@ -0,0 +1,61 @@
+
+## Requirements
+
+Similar to [BatchRequestContent](BatchRequestContent.md), it must support the below functionality while holding a collection of BatchRequestContent instances.
+
+- It MUST be possible to add native platform HTTP request objects to a BatchRequestContent object. The request URL MUST be a relative URL. Adding Graph specific Request objects MAY be supported.
+- Added requests MUST be assigned unique identifiers. Those identifiers MAY be user provided or autogenerated. The uniqueness SHOULD be enforced/validated across the same `BatchRequestContent` instance the request objects are held in AND across the `BatchRequestContentCollection` instance.
+- It MUST be possible to create dependencies between request objects. However, these dependencies SHOULD be enforced/validated across the same `BatchRequestContent` instance the request objects are held in and NOT across `BatchRequestContent` instances in the `BatchRequestContentCollection`.
+- The addition of a new request object with a dependency, SHOULD make the best effort at placing the object in the same `BatchRequestContent` instance as its dependency and throw an error in the event this is not possible.
+- It MUST be possible to configure the maximum number of request objects per `BatchRequestContent` in the `BatchRequestContentCollection` to a value lower than the default value of 20 request objects. This is to enable customized scenarios for developers and provide a means of client side optimization/throttling for workloads that may have lower maximums in batched requests.
+- It MUST possible to use a dictionary of existing unique identifiers and their corresponding to response status code to create a new/fresh `BatchRequestContentCollection` instance to unlock re-emitting/retrying failed requests from an existing `BatchRequestContentCollection` instance.
+
+> Given the similar API experience between `BatchRequestContentCollection` and `BatchRequestContent`, public documentation(code comments and examples) SHOULD encourage the use and point to `BatchRequestContentCollection` to avoid confusion for the end user. Adding new requests should automatically generate new `BatchRequestContent` instances to the `BatchRequestContentCollection`. `BatchRequestContent` should be a public facing type to enable users with customized batching scenarios.
+
+## Performance Considerations
+
+To minimize the risk of clients getting throttled by the API, the request client SHOULD send `BatchRequestContent` instances in sequence if the `BatchRequestContentCollection` contains more than one `BatchRequestContent` instance by default. The request client MAY send `BatchRequestContent` instances in parallel by use of a configuration parameter to allow the user to opt-in to this behavior in the event that the user is aware of the potential throttling limits.
```

Comment:
I'd love @darrelmiller to chime in here on the sequential vs. parallel execution of batches. I understand it seems like the right approach, but I'd love a perspective from the inside to ensure we're doing the right thing.

Reply:
It would be awesome for our "batch" mechanism to have the ability to parallelize some of the requests.  This is especially the case considering Graph has a limit of 20 requests per batch.  There was a performance analysis done on calling Graph a few years ago by Peter C. at one of the Build sessions that showed how you could combine batches and parallel requests to get the best performance.

Reply:
If you know any paper / approach we should be taking @darrelmiller, please share them!